cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -888,6 +888,31 @@ Slice::CsGenerator::writeUnmarshalCode(Output &out,
     }
 }
 
+namespace
+{
+bool isVariableLength(TypePtr type)
+{
+    BuiltinPtr builtin = BuiltinPtr::dynamicCast(type);
+    ProxyPtr proxy = ProxyPtr::dynamicCast(type);
+
+    if(builtin &&
+       builtin->kind() != Builtin::KindValue &&
+       builtin->kind() != Builtin::KindObjectProxy &&
+       builtin->kind() != Builtin::KindString)
```

Comment:
There's a `builtin->isVariableLength()`method to simplify checks like this, that might be applicable here:
https://github.com/zeroc-ice/ice/blob/038f547d2d1558943b8efd97f359f01c280b42c1/cpp/src/Slice/Parser.cpp#L508
One difference however is that it will return `true` for `Builtin::KindObject`, but given that we're planning on merging this with `Builtin::KindObjectProxy`, this might not be a concern?

Reply:
Nevermind, looks like Bentoi beat me to it!